src/core/taint.c

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <sys/utsname.h>
+
+#include "alloc-util.h"
+#include "clock-util.h"
+#include "cgroup-util.h"
+#include "errno-util.h"
+#include "fileio.h"
+#include "fs-util.h"
+#include "log.h"
+#include "os-util.h"
+#include "path-util.h"
+#include "strv.h"
+#include "taint.h"
+#include "uid-range.h"
+
+static int short_uid_gid_range(bool uid) {
+        _cleanup_(uid_range_freep) UIDRange *p = NULL;
+        int r;
+
+        /* Taint systemd if we the UID range assigned to this environment doesn't at least cover 0…65534,
+         * i.e. from root to nobody. */
+
+        const char *path = uid ? "/proc/self/uid_map" : "/proc/self/gid_map";
```

Comment:
Well, I mean that `uid_range_load_userns` defaults to `/proc/self/uid_map` or `/proc/self/gid_map` when `UID_RANGE_USERNS_INSIDE` or `GID_RANGE_USERNS_INSIDE` is specified, respectively, and path can be left out in that case

Reply:
OK, updated again.